exporter/lightstepexporter/lightstep.go

```diff
@@ -0,0 +1,77 @@
+// Copyright 2019 OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package lightstepexporter
+
+import (
+	"context"
+
+	"github.com/lightstep/opentelemetry-exporter-go/lightstep"
+	"github.com/open-telemetry/opentelemetry-collector/consumer/consumerdata"
+	"github.com/open-telemetry/opentelemetry-collector/exporter"
+	"github.com/open-telemetry/opentelemetry-collector/exporter/exporterhelper"
+	"github.com/open-telemetry/opentelemetry-collector/oterr"
+)
+
+// LightStepExporter contains a pointer to a LightStep opentelemetry exporter.
+type LightStepExporter struct {
+	exporter *lightstep.Exporter
+}
+
+func newLightStepTraceExporter(cfg *Config) (exporter.TraceExporter, error) {
+	exporter, err := lightstep.NewExporter(
+		lightstep.WithAccessToken(cfg.AccessToken),
+		lightstep.WithHost(cfg.SatelliteHost),
+		lightstep.WithPort(cfg.SatellitePort),
+		lightstep.WithServiceName(cfg.ServiceName),
+	)
+
+	if err != nil {
+		return nil, err
+	}
+
+	lsExporter := LightStepExporter{
+		exporter: exporter,
+	}
+
+	return exporterhelper.NewTraceExporter(
+		cfg,
+		lsExporter.pushTraceData,
+		exporterhelper.WithTracing(true),
+		exporterhelper.WithMetrics(false),
```

Comment:
nit: this should be true so it generates metrics for the exporter (this option is going to be removed soon to avoid such mistakes, I will take care of that when it reaches the repo)
```suggestion
		exporterhelper.WithMetrics(true),
```